### DISCUSSION TITLE
Popover re-calculates position when content size changes

### DISCUSSION THREAD

**Discussion by philipaarseth** [radix-ui/primitives#1256] [Help]
Hi,

Is there a reason why radix popovers don't simply let CSS recalculate the element position with a `translateX(-50%)` instead of re-calculating and updating top & left?

Also, why is it not bottom oriented if side is top? 

Unless I'm missing something, this approach will lead to UI shake like so:

![recalculating position](https://user-images.githubusercontent.com/31098164/159027881-6461aee5-48ad-47be-a669-fbe8241ae326.gif)

**Comment by benoitgrelard** [comment]
Hi @philipaarseth, our popper solution will recalculate positioning "live" based on any change in position/size for the target (trigger here) and size for the content.

> Also, why is it not bottom oriented if side is top?

Not sure I understand sorry.

> Unless I'm missing something, this approach will lead to UI shake like so:

If you could create a quick sandbox for reproduction, I'm happy to take a look.
As far as I can tell on my side, I don't see the issue when the height changes:

https://user-images.githubusercontent.com/1539897/160803799-32b3078d-43c9-4396-a900-7fc674cd4f68.mov
